vit-servicing-station-tests/src/tests/non_functional/mod.rs

```diff
@@ -1,53 +1,150 @@
 use crate::common::{clients::RestClient, startup::quick_start};
 use assert_fs::TempDir;
 
+use crate::common::clients::graphql::GraphqlClient;
+use crate::common::data::Snapshot;
 use jortestkit::load::{self, Configuration, Monitor, Request, RequestFailure};
 use rand_core::{OsRng, RngCore};
 
 #[derive(Clone, Debug)]
 struct VitRestRequest {
+    snapshot: Snapshot,
     rest_client: RestClient,
+    graphql_client: GraphqlClient,
     random: OsRng,
 }
 
-impl From<RestClient> for VitRestRequest {
-    fn from(rest_client: RestClient) -> Self {
+impl VitRestRequest {
+    pub fn new(
+        snapshot: Snapshot,
+        mut rest_client: RestClient,
+        mut graphql_client: GraphqlClient,
+    ) -> Self {
+        rest_client.disable_log();
+        graphql_client.disable_log();
+
         Self {
+            snapshot,
             rest_client,
+            graphql_client,
             random: OsRng,
         }
     }
+
+    fn random_token(&self) -> String {
+        let tokens = self.snapshot.tokens();
+        let random_idx = self.radnom_usize() % tokens.len();
+        tokens
+            .keys()
+            .cloned()
+            .collect::<Vec<String>>()
+            .get(random_idx)
+            .unwrap()
+            .clone()
+    }
+
+    fn radnom_usize(&self) -> usize {
+        self.random.clone().next_u32() as usize
+    }
+
+    fn random_proposal_id(&self) -> i32 {
+        let proposals = self.snapshot.proposals();
+        let random_idx = self.radnom_usize() % proposals.len();
+        proposals.get(random_idx).unwrap().internal_id.clone()
+    }
+    fn random_fund_id(&self) -> i32 {
+        let funds = self.snapshot.funds();
+        let random_idx = self.radnom_usize() % funds.len();
+        funds.get(random_idx).unwrap().id.clone()
+    }
 }
 
 impl Request for VitRestRequest {
     fn run(&self) -> Result<(), RequestFailure> {
-        match self.random.clone().next_u32() % 2 {
-            0 => self
-                .rest_client
+        let random_token = self.random_token();
```

Comment:
This would look like : `let random_token = RandomMod::random_token(self.snapshot.tokens())`